op-acceptance-tests/tests/base/chain_test.go

```diff
@@ -1,48 +1,97 @@
 package base
 
 import (
+	"context"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
-
 	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
 	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
 	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 )
 
 // TestChainFork checks that the chain does not fork.
 func TestChainFork(t *testing.T) {
-	chainIdx := uint64(0)
-
 	systest.SystemTest(t,
-		chainForkTestScenario(chainIdx),
+		chainForkTestScenario(),
 	)
 }
 
-func chainForkTestScenario(chainIdx uint64) systest.SystemTestFunc {
+func chainForkTestScenario() systest.SystemTestFunc {
 	return func(t systest.T, sys system.System) {
-		const numCyclesToWait = 5
-
 		logger := testlog.Logger(t, log.LevelInfo)
 		logger.Info("Started test")
 
-		chain := sys.L2s()[chainIdx]
+		// Check all L2 chains
+		for i, chain := range sys.L2s() {
+			chainIndex := i
+			currentChain := chain
+			t.Run(fmt.Sprintf("Chain_%d", chainIndex), func(t systest.T) {
+				t.Parallel()
+				chainLogger := logger.New("chain", chainIndex)
+
+				// Initial chain fork check
+				laterCheck, err := systest.CheckForChainFork(t.Context(), currentChain, chainLogger)
+				if err != nil {
+					t.Fatalf("first chain fork check failed: %v", err)
+				}
+
+				// Get a geth client
+				chainLogger.Info("Waiting for next block")
+				node := currentChain.Nodes()[0]
+				client, err := node.GethClient()
+				if err != nil {
+					t.Fatalf("failed to get geth client: %v", err)
+				}
+
+				// Create a context with timeout
+				ctx, cancel := context.WithTimeout(t.Context(), 60*time.Second)
+				defer cancel()
+
+				// Get the current block number
+				currentBlockNum, err := client.BlockNumber(t.Context())
+				if err != nil {
+					t.Fatalf("failed to get current block number: %v", err)
+				}
+
+				targetBlockNum := currentBlockNum + 1
+				chainLogger.Debug("Waiting for block", "target", targetBlockNum)
```

Comment:
Any reason not to just use `wait.ForBlock`?
https://github.com/ethereum-optimism/optimism/blob/c9b56cd11a74f07d8a3c9f770b2fd1d542b2b57a/op-e2e/e2eutils/wait/blocks.go#L24

Reply:
Oh, nice, I wasn't aware of that utility function. I'll switch to using that. Thanks!